docs/rules/prefer-query-by-disappearance.md

```diff
@@ -0,0 +1,33 @@
+# Suggest using `queryBy*` queries when waiting for disappearance (`testing-library/prefer-query-by-disappearance`)
+
+## Rule Details
+
+This rule enforces using `queryBy*` queries when waiting for disappearance with `waitForElementToBeRemoved`.
+
+Using `queryBy*` queries in a `waitForElementToBeRemoved` yield more descriptive error messages and helps achieving more consistency in a codebase.
```

Comment:
```suggestion
Using `queryBy*` queries in a `waitForElementToBeRemoved` yields more descriptive error messages and helps to achieve more consistency in a codebase.
```

minor spelling fixes (I'm not an English native speaker though, but Grammarly is marking these as errors 😅 )